tests/cypress/support/commands_filters_feature.js

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2021 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+
+/// <reference types="cypress" />
+
+Cypress.Commands.add('сheckFiltersModalOpened', () => {
+    cy.document().then((doc) => {
+        const filterModal = Array.from(doc.querySelectorAll('.cvat-filters-modal-visible'));
+        if (filterModal.length === 0) {
+            cy.contains('.cvat-annotation-header-button', 'Filters').click();
+        }
+    });
+});
+
+Cypress.Commands.add('collectGroupID', () => {
+    let groupDataID = [];
+    cy.get('.group').then(($group) => {
+        for (let i = 0; i < $group.length; i++) {
+            groupDataID.push($group[i].dataset.id);
+        }
+        return groupDataID;
+    });
+});
+
+Cypress.Commands.add('collectRuleID', () => {
+    let ruleDataID = [];
+    cy.get('.rule').then(($rule) => {
+        for (let i = 0; i < $rule.length; i++) {
+            ruleDataID.push($rule[i].dataset.id);
+        }
+        return ruleDataID;
+    });
+});
+
+Cypress.Commands.add('clearFilters', () => {
+    cy.сheckFiltersModalOpened();
+    cy.contains('button', 'Clear filters').click();
+});
+
+Cypress.Commands.add('addFiltersGroup', (groupIndex) => {
+    cy.сheckFiltersModalOpened();
+    cy.collectGroupID().then((groupIdIndex) => {
+        cy.get(`[data-id="${groupIdIndex[groupIndex]}"]`).contains('button', 'Add group').first().click();
+    });
+});
+
+Cypress.Commands.add('addFiltersRule', (groupIndex) => {
+    cy.сheckFiltersModalOpened();
+    cy.collectGroupID().then((groupIdIndex) => {
+        cy.get(`[data-id="${groupIdIndex[groupIndex]}"]`).contains('button', 'Add rule').click();
+    });
+});
+
+Cypress.Commands.add('setGroupConition', (groupIndex, contidion) => {
+    cy.сheckFiltersModalOpened();
+    cy.collectGroupID().then((groupIdIndex) => {
+        cy.get(`[data-id="${groupIdIndex[groupIndex]}"]`).within(() => {
+            cy.get('.group--header').first().trigger('mouseover');
+            cy.contains('button', contidion).click({ force: true });
+        });
+    });
+});
+
+Cypress.Commands.add(
+    'setFilter',
+    (groupIndex, ruleIndex, field, operator, valueSource, value, label, labelAttr, submit) => {
```

Comment:
Too many arguments. Can we use object with destructurization instead? 
```({groupIndex, ruleIndex, field, operator, valueSource, value, label, labelAttr, submit})```

cy.setFilter({groupIndex, ruleIndex, field, operator, valueSource, value, label, labelAttr, submit})

Reply:
Done. Reworked the test also.